dragonflybsd/testing-theory/dragonflybsd_regression_Q&A.md

```diff
@@ -0,0 +1,81 @@
+# DragonflyBSD tests research
+
+## Releases
+
+**How is OpenBSD released?**
+
+Releases occur approximately twice a year. DragonFly release branches only
+contain bug and security fixes and are designed for people running production
+systems who don't want any surprises. Brand-new features often discussed on
+the mailing lists are typically not in release branches.
+
+DragonFly systems based on releases are labeled RELEASE, for example, you
+might be running DragonFly X.Y.Z-RELEASE. If you run a daily snapshot or
+track git master, you'll see X.Y.Z-DEVELOPMENT as your system version.
+
+## Regression repository
+
+**How are regression tests written (language, transparency, syntax)?**
+
+Tests are mainly written in C/C++ with support of some files with various
+formats.
+
+## OS installation in automated test
+
+1. Automatic os installation test are currently implemented in various cases
+in our testing env (eg. pfSense, Ubuntu, Debian). They're implemented using
+Robot Framework and Bash scripts. Main issue would be creating unattended
+installer __and documentation how to create one__.
+
+[Auto installer creation guide](https://umbriel.fr/blog/DragonFly_BSD_autoinstall.html)
+[Similar guide for FreeBSD](https://www.freebsd.org/cgi/man.cgi?bsdinstall(8))
+
+## Contributing to the regression repository
+
+DragonflyBSD doesn't have contribution guidelines. It may be necessary to
+contact maintainers for information.
+
+## Test scope
+
+**What tests should be added?**
+
+All functionalities that will be added should have tests. More it will be
+possible to describe in detail after the implementation is completed.
+
+What we would probably like to check:
+
+Values ​​from the ESRT table, however, a tool from
+[this site](https://reviews.freebsd.org/rG24f398e7a153a05a7e94ae8dd623e2b6d28d94eb)
```

Comment:
@PrzemyslawBanasiak13 AFAIK we do not have in plants to create additional similar to `efitable` tool. In DragonflyBSD we want to use `kenv`

Reply:
@kotylamichal So functionalities like `efi_append_variable()`will not be added?

Reply:
@PrzemyslawBanasiak13 functions like `efi_append_variable()` will be added in `efivar` - am I correct @SergiiDmytruk ? Custom tool like `efitable` from https://reviews.freebsd.org/rG24f398e7a153a05a7e94ae8dd623e2b6d28d94eb will not be used in DragonFly, Open, and NetBSD

@TomaszAIR cc

Reply:
> functions like `efi_append_variable()` will be added in `efivar`

Yes, it's an API of `libefivar`.

> Custom tool like `efitable` from https://reviews.freebsd.org/rG24f398e7a153a05a7e94ae8dd623e2b6d28d94eb will not be used in DragonFly, Open, and NetBSD

Probably won't be upstreamed, but I did [adapt it](https://github.com/3mdeb/openbsd-src/commit/ae73eed8edd52b851b3fabc76f5e3701aca4a095) for OpenBSD and it should work for other BSDs with small changes. Upstream is unlikely to keep `dmesg` output and this tool can be used to verify that the API works. Similarly for EFI vars, I made [another single-file tool](https://github.com/3mdeb/openbsd-src/commit/c1a882abc82fc46d5d66fde4794fcf9bee000fe1) which is easy to build and it can used for tests as well. `efivar` from RedHat has `efivar` binary of its own, but it's much bigger, has dependencies and needs patches to work on BSDs.

Reply:
I've added a section informing that we will be using `kenv` however, I'm not deleting anything as it may be useful anyway